client/src/legacy/LeftAndMain.EditForm.js

```diff
@@ -566,22 +566,74 @@ $.entwine('ss', function($){
     onmatch: function () {
       if (this.val() === 'OnlyTheseUsers') {
         if (this.is(':checked')) {
-          this.showList(true);
+          this.showGroupsList(true);
+          this.hideMembersList(true);
         } else {
-          this.hideList(true);
+          this.hideGroupsList(true);
+        }
+      }
+
+      if (this.val() === 'OnlyTheseMembers') {
+        if (this.is(':checked')) {
+          this.showMembersList(true);
+          this.hideGroupsList(true);
+        } else {
+          this.hideMembersList(true);
         }
       }
     },
     onchange: function (e) {
       if (e.target.value === 'OnlyTheseUsers') {
-        this.showList();
+        this.showGroupsList();
+        this.hideMembersList();
+      } else {
+        this.hideGroupsList();
+      }
+      if (e.target.value === 'OnlyTheseMembers') {
+        this.showMembersList();
+        this.hideGroupsList();
+      } else {
+        this.hideMembersList();
+      }
+    },
```

Comment:
This is effectively the exact same code twice - can you please create a new function (maybe "toggleListDisplay" or similar) that takes a value as an argument, and sets the display accordingly?
For `onChange` the value is obviously `e.target.value`. For `onmatch` the value I think is just whichever of the two options `is(':checked')`